crates/oxc_linter/src/rules/typescript/no_explicit_any.rs

```diff
@@ -0,0 +1,211 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+"typescript-eslint(no-explicit-any): Using any disables many type checking rules and is generally best used only as a last resort or when prototyping code"
+)]
+#[diagnostic(severity(warning))]
+struct NoExplicitAnyDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoExplicitAny {
+    pub ignore_rest_args: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow the any type.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The any type in TypeScript is a dangerous "escape hatch" from the type system.
+    /// Using any disables many type checking rules and is generally best used only as a last
+    /// resort or when prototyping code. This rule reports on explicit uses of the any keyword
+    /// as a type annotation.
+    ///
+    /// ### Example
+    /// ```typescript
+    /// const age: any = 'seventeen';
+    /// const ages: any[] = ['seventeen'];
+    /// const ages: Array<any> = ['seventeen'];
+    /// ```
+    NoExplicitAny,
+    correctness
+);
+
+impl Rule for NoExplicitAny {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if self.ignore_rest_args {
+            // 
+        } else {
+            if let AstKind::TSAnyKeyword(decl) = node.kind() {
+                ctx.diagnostic(NoExplicitAnyDiagnostic(decl.span));
+            }
+        }
+        
+    }
+
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let ignore_rest_args = value
+            .get(0)
+            .and_then(|config| config.get("ignoreRestArgs"))
+            .and_then(serde_json::Value::as_bool)
+            .unwrap_or(false);
+
+        Self { ignore_rest_args }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "const number: number = 1;",
```

Comment:
todo: use tuples with configuration options for test cases. See: https://github.com/Boshen/oxc/blob/main/crates/oxc_linter/src/rules/eslint/getter_return.rs#L189